astropy/io/ascii/ui.py

```diff
@@ -593,6 +621,118 @@ def _get_guess_kwargs_list(read_kwargs):
     return guess_kwargs_list
 
 
+def _read_in_chunks(table, **kwargs):
+    """
+    For fast_reader read the ``table`` in chunks and vstack to create
+    a single table, OR return a generator of chunk tables.
+    """
+    fast_reader = kwargs['fast_reader']
+    chunk_size = fast_reader.pop('chunk_size')
+    chunk_generator = fast_reader.pop('chunk_generator', False)
+    fast_reader['parallel'] = False  # No parallel with chunks
+
+    tbl_chunks = _read_in_chunks_generator(table, chunk_size, **kwargs)
+    if chunk_generator:
+        return tbl_chunks
+
+    tbl0 = next(tbl_chunks)
+    masked = tbl0.masked
+
+    # Numpy won't allow resizing the original so make a copy here.
+    out_cols = {col.name: col.data.copy() for col in tbl0.itercols()}
+
+    for tbl in tbl_chunks:
+        col_len = len(tbl)
+        masked |= tbl.masked
+        for name, col in tbl.columns.items():
+            if masked:
+                # For masked case, concatenate old + new columns
+                out_cols[name] = np.ma.concatenate([out_cols[name], col.data])
```

Comment:
why not just use `np.ma.resize` here for symmetry?
```
out_col = out_cols[name]
resize = np.resize if ~masked else np.ma.resize
outcol = resize(len(out_col) + col_len, refcheck=False)
```
```

Reply:
I tried that first, but (annoyingly) a masked array does not "own" its data so resizing in-place is simply not allowed.